File: website/init_db.py
# init_db.py
import os
from app import app, db
from app.models import User

def init_db():
    db_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'app.db')
    
    # Remove existing database file if it exists
    if os.path.exists(db_path):
        os.remove(db_path)
        print("Existing database file removed.")
    
    # Create new database tables
    with app.app_context():
        db.create_all()
        print("Database tables created.")

if __name__ == "__main__":
    init_db()

File: goserver/init_db.go
package main

import (
	"log"
	"os"

	"boquila/app/models"

	"gorm.io/driver/sqlite"
	"gorm.io/gorm"
)

func main() {
	dbPath := "app.db"

	// Remove existing database file if it exists
	if _, err := os.Stat(dbPath); err == nil {
		os.Remove(dbPath)
		log.Println("Existing database file removed.")
	}

	// Create new database
	db, err := gorm.Open(sqlite.Open(dbPath), &gorm.Config{})
	if err != nil {
		log.Fatalf("failed to connect to the database: %v", err)
	}

	// Auto-migrate database models
	db.AutoMigrate(&models.User{}, &models.Challenge{})
	log.Println("Database tables created.")
}
